beacon_node/http_api/src/attester_duties.rs

```diff
@@ -20,16 +23,32 @@ pub fn attester_duties<T: BeaconChainTypes>(
     let current_epoch = chain
         .epoch()
         .map_err(warp_utils::reject::beacon_chain_error)?;
-    let next_epoch = current_epoch + 1;
 
-    if request_epoch > next_epoch {
+    // Determine what the current epoch would be if we fast-forward our system clock by
+    // `MAXIMUM_GOSSIP_CLOCK_DISPARITY`.
+    //
+    // Most of the time, `tolerant_current_epoch` will be equal to `current_epoch`. However, during
+    // the first `MAXIMUM_GOSSIP_CLOCK_DISPARITY` duration of the epoch `tolerant_current_epoch`
+    // will equal `current_epoch + 1`
+    let tolerant_current_epoch = chain
+        .slot_clock
+        .now_with_future_tolerance(MAXIMUM_GOSSIP_CLOCK_DISPARITY)
+        .ok_or_else(|| warp_utils::reject::custom_server_error("unable to read slot clock".into()))?
+        .epoch(T::EthSpec::slots_per_epoch());
+
+    if request_epoch == current_epoch
```

Comment:
Don't think this makes a functional difference unless `MAXIMUM_GOSSIP_CLOCK_DISPARITY` is greater than an epoch (lol) but this logic makes a bit more sense to me: 
 
```
if current_epoch <= request_epoch && request_epoch <= tolerant_current_epoch + 1 {
} else if request_epoch > tolerant_current_epoch + 1 {
} else {
// request < current_epoch
}
```

Reply:
Agree it should be equivalent, but don't have a strong preference either way. The 4-way _or_ has the advantage of being quite explicit.

Writing `request_epoch >= current_epoch && request_epoch <= tolerant_current_epoch + 1`  is also nice :stuck_out_tongue_closed_eyes:

Reply:
Thanks @realbigsean, I agree that your statement is more universal but I do find what we've got to be easier to read. Perhaps that's just because I wrote it :thinking:. I might leave it as-is for now :pray: